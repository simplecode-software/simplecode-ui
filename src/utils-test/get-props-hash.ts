import { TUnknownObject } from '../types';

const asString = (value: unknown): string => {
    if (typeof value === 'string') {
        return `"${value}"`;
    }
    if (typeof value === 'number') {
        return `${value}`;
    }
    if (typeof value === 'boolean') {
        return `${value ? 'true' : 'false'}`;
    }
    if (typeof value === 'function') {
        return '[f]';
    }
    if (value instanceof Error) {
        return `"${value.message || value.constructor.name}"`;
    }
    return JSON.stringify(value);
};


const getHash = (input: string): string => {
    let hash: number = 0;
    for (let i = 0; i < input.length; i++) {
        // eslint-disable-next-line no-bitwise
        hash = input.charCodeAt(i) + ((hash << 5) - hash);
    }
    return hash.toString(16);
};

export const getPropsHash = <T extends TUnknownObject>(props: T): string => {
    const propStrings = Object.entries(props).map(([key, value]) => {
        if (value === null || value === undefined) {
            return '';
        }
        return `${key}={${asString(value)}}`;
    });
    const plainText = propStrings.filter((x) => x).join(',');
    
    return plainText ? getHash(plainText).replace(/^-/, '0') : '[EMPTY]';
};
