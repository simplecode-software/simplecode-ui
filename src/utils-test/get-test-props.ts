import { emptyArray, emptyObject } from '../constants/empty';
import { TPropVariants } from '../types';


export const getTestProps = <T extends Record<string, unknown>>(
    variants: TPropVariants<T>,
): ReadonlyArray<T> => {
    const keys: ReadonlyArray<keyof T> = Object.keys(variants);
    const values: ReadonlyArray<ReadonlyArray<T>> = Object.values(variants);
    
    const maxVariants = Math.max.apply(null, values.map((item) => item.length));
    
    let testProps: ReadonlyArray<T> = emptyArray as ReadonlyArray<T>;
    
    for (let i = 0; i < maxVariants; i++) {
        let props = emptyObject as T;
        keys.forEach((key) => {
            const values = variants[key];
            if (!values) {
                return;
            }
            const index = i % values.length;
            props = { ...props, [key]: values[index] };
        });
        
        testProps = [...testProps, props];
    }
    
    return testProps;
};
