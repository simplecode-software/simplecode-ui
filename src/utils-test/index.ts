export * from './get-props-hash';
export * from './get-test-props';
export * from './test-samples';
export * from './test-snapshots';
