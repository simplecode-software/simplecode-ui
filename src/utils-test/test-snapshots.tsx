import React from 'react';
import renderer from 'react-test-renderer';

import { TestRenderer } from '../../src-web';
import { TAnyComponent, TPropVariants } from '../types';
import { getPropsHash } from './get-props-hash';
import { getTestProps } from './get-test-props';


jest.mock('../../src-web/UiProviderWeb/ToastWeb/ToastWeb');


export const testSnapshots = <T extends Record<string, unknown>>(
    name: string,
    Component: TAnyComponent,
    variants: TPropVariants<T>,
): void => {
    describe(name, () => {
        getTestProps(variants).forEach((props) => {
            it(`renders ${getPropsHash(props)}`, () => {
                const component = renderer.create((
                    <TestRenderer>
                        <Component {...props} />
                    </TestRenderer>
                ));
                expect(component.toJSON()).toMatchSnapshot();
            });
        });
    });
};
