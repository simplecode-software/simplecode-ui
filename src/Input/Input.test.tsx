import React from 'react';

import { testSnapshots, samplePropsInput } from '../utils-test';
import { Input } from './Input';


jest.mock('../Flex/Flex');
jest.mock('../ListItem/ListItem');
jest.mock('../InputZone/InputZone');

testSnapshots('spec/Input', Input, samplePropsInput);
