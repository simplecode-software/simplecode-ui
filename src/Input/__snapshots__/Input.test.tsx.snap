// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spec/Input renders [EMPTY] 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 02df4b4b85 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 03f68f5d5a 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 05ba45c587 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 05d680b291 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 06f940b014 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 07c6d20965 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 08cfbf8a22 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 013c4a723da 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 050df530df 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 075bbe81cf 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0458bf1380 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0872bd1793 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 016410da1bf 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0a00e7f70 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0a0a10d56f 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0b2f20bbb0 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0e2b538012 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 0fae36c43d 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 2eb88069a 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 12b5bc0f11 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 17bb0cf0b 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 28eb3b872 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 30a7d2440 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 66e8b53ac 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 83ed6ecbe 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 91aefc0b2 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 447db8734 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 665e86a63 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 24517f6b3 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders 81384c77c 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;

exports[`spec/Input renders c08f3e0d4 1`] = `
<test-mock
  events="auto"
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <test-mock
    style={
      [
        {
          "bottom": 0,
          "height": undefined,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        },
        {
          "pointerEvents": "none",
          "position": "fixed",
          "zIndex": 1000,
        },
      ]
    }
  >
    <test-mock
      align="flex-end"
      events="box-none"
      justify="flex-start"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "bottom": 100,
            "zIndex": 100,
          },
        ]
      }
    />
  </test-mock>
</test-mock>
`;
